### PR TITLE
Specified directories for routes, views and jobs tests

### PR DESCRIPTION
**Pull request by go2null**
### Summary

Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together.
### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Finally, if your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](http://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

Thanks for contributing to Rails!
- Added location for route, view and job tests to section 2.1.  (They are currently only defined in sections 8 Testing Views and 11 Testing Jobs.)
- Added location for route test in section 7 Testing Routes.  (Currently only defined in section 8 Testing Views.)

### CODE REVIEW

File: guides/source/testing.md [hunk 1]
```diff
@@ -37,10 +37,12 @@ controllers/    helpers/        mailers/        test_helper.rb
 fixtures/       integration/    models/
 ```
 
-The `models` directory is meant to hold tests for your models, the `controllers` directory is meant to hold tests for your controllers and the `integration` directory is meant to hold tests that involve any number of controllers interacting. There is also a directory for testing your mailers and one for testing view helpers.
+The `models` directory is meant to hold tests for your models, the `controllers` directory is meant to hold tests for your controllers, routes and views and the `integration` directory is meant to hold tests that involve any number of controllers interacting. There is also a directory for testing your mailers and one for testing view helpers.
```
Comment (eileencodes) [inline guides/source/testing.md]: We prefer the oxford comma in documentation so change "tests for your controllers, routes and views" to "tests for your controllers, routes, and views" to have the comma after routes.

Additionally this change has created a bit of a run-on sentence. Can you change the first sentence to be 2 sentences, like so:

> The `models` directory is meant to hold tests for your models, the `controllers` directory is meant to hold tests for your controllers, routes, and views. The `integration` directory is meant to hold tests that involve any number of controllers interacting

**Comment by go2null** [comment]
How about adding a semicolon to separate the parts,as they are pretty much independent clauses.

Alternative 1 - with semicolon:
The `models` directory is meant to hold tests for your models; the `controllers` directory is meant  to hold tests for your controllers, routes, and views; and the `integration` directory is meant to hold tests that involve any number of controllers interacting. There is also a directory for testing your `mailers` and another one for testing view `helpers`.

Alternative 2 - with separate sentences:
The `models` directory is meant to hold tests for your models. The `controllers` directory is meant to hold tests for your controllers, routes, and views. The `integration` directory is meant to hold tests that involve any number of controllers interacting. There is also a directory for tests for your `mailers` and another one for testing view `helpers`.

Alternative 3 - rewrite:
The `helpers`, `mailers`, and `models` directories are meant to hold tests for your view helpers, mailers, and and models, respectively. The `controllers` directory is meant to hold tests for your controllers, routes, and views. The `integration` directory is meant for tests interactions between controllers.

**Comment by eileencodes** [comment]
I like the last one best. Thanks!

The helpers, mailers, and models directories are meant to hold tests for view helpers, mailers, and models, respectively. The controllers directory is meant to hold tests for controllers, routes, and views. The integration directory is meant to hold tests for interactions between controllers.

**Comment by eileencodes** [comment]
Thanks! Can you squash your commits into one?

**Comment by go2null** [comment]
Rebased to a single commit.

**Comment by eileencodes** [comment]
Thank you so much @go2null! 😄

**Pull request merged**